Newsletter vol.4 (#4340)
line/armeria: site/src/pages/news/20220708-newsletter-4.mdx

---
date: 2022-07-08
---

# Armeria Newsletter vol. 4

<video class="hideOnReducedMotion" src="../../images/armeria.m4v" preload="none" autoplay muted loop style="width: 282px; height: 112px;">
  <img src="../../images/armeria.gif" loading="lazy" width="282" height="112" />
</video>

## From the devs

<div class="noBullet">

> - ✨ We released several releases since our last newsletter with a bunch of new exciting features!

Reviewer: According to [Amazon's writing style tips](https://medium.com/fact-of-the-day-1/amazon-writing-style-tip-2-2f77bb3bbbb0), it recommends replacing adjectives with numbers. 😆
```suggestion
- ✨ We've released 7 new versions since our last newsletter with a bunch of new exciting features!
```

  - [1.13.4](https://armeria.dev/release-notes/1.13.4/)
    - Security fix(CWE-22)
  - [1.14.0](https://armeria.dev/release-notes/1.14.0/)
    - Experimental support for Scala 3
    - Fluent conversion of <type://HttpResponse> into a desired type using <type://WebClient>
    - Various build options for gRPC using <type://GrpcClientBuilder>
  - [1.14.1](https://armeria.dev/release-notes/1.14.1/)
    - Additional logs for <type://EmptyEndpointGroupException>
  - [1.15.0](https://armeria.dev/release-notes/1.15.0/)

>     - User-friendly gRPC health check protocol

Reviewer 2: I'm not sure if this additional support is particularly use friendly

```suggestion
    - Support for gRPC health check protocol
```

    - Multipart file uploads in annotated services

>     - Preliminary GraphQL support in <type://DocService>

Reviewer 2: Since preliminary sounds like it's not ready for usage, but it is 😄 

```suggestion
    - GraphQL support in <type://DocService>
```

>     - Refactored DMS resolver cache with higher hit ratio and reduced traffic

Reviewer 2: ```suggestion
    - Refactored DNS resolver cache with higher hit ratio and reduced traffic
```

    - New metrics that help you get alerts before your TLS certificates are expired
  - [1.16.0](https://armeria.dev/release-notes/1.16.0/)
    - New metrics for TLS handshake results
    - Handling <type://Decorator> annotation with <type://GrpcService> for your gRPC stubs
    - Programmatic <type://Flags> configuration via Java SPI
  - [1.17.0](https://armeria.dev/release-notes/1.17.0/)
    - Easier way to send and receive RESTful APIs
    - Ability to limit the maximum length of Thrift messages
    - Injecting dependencies in annotations
    - Support for Kotlin nullable type(`?`) in annotated services

</div>
<div class="noBullet">

- 👂 Armeria team is hosting [bi-weekly virtual office hours](https://armeria.dev/community/office-hours/).

>   - It was great talking with all of you through the video chat. And we're still listening. :)

Reviewer 2: I think the sentence sounds a little vague. Did you mean we're still doing virtual hours?

```suggestion
  - It was great talking with all of you through video chat. We hope we can see you again in the near future. :)
```

</div>

## From the community

<div class="noBullet">

> - 🎓 Try out this guided tutorial for REST Service](https://armeria.dev/tutorials/rest/blog)! It's super simple, and super friendly to understand building a RESTful service with Aremria. This tutorial will guide you to make a blog service, with which you can create, read, update and delete blog posts. In particular, you'll be using these Armeria features:

Reviewer 2: - Shouldn't the link be on `tutorial` instead of `REST Service`?
- Prefer active voice, "It" is vague in this context
- Removed a redundant sentence

```suggestion
- 🎓 Try out this guided [tutorial](https://armeria.dev/tutorials/rest/blog) for creating a RESTful service! This super friendly and simple tutorial will guide you to make a blog service, with which you can create, read, update and delete blog posts. In particular, you'll be using these Armeria features:
```

  - Service annotations
  - Request converter
  - Parameter injection
  - Exception handler
  - Blocking task executor
- 🦾 MSA(Microservices Architecture): Harder, Better, Faster, Stronger

>   - In the session, @cnabro introduces how the "Series On" service converted monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentioned about how they monitored microservices using Zipkin and Elastic Search.

Reviewer 3: ```suggestion
  - In the session, @cnabro introduces how [Series On](https://serieson.naver.com/) migrated their monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentioned how they monitored microservices using [Zipkin](https://zipkin.io/) and [Elasticsearch](https://www.elastic.co/).
```

>   - In the session, @cnabro introduces how the "Series On" service converted monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentioned about how they monitored microservices using Zipkin and Elastic Search.

Reviewer 2: The tense should be consistent (introduces <-> mentions)

```suggestion
  - In the session, @cnabro introduces how the "Series On" service converted monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentions how they monitored microservices using Zipkin and Elastic Search.
```

>   - Here are the [Video](https://youtu.be/1TsGYR1dves) and [Slide](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))

Reviewer 2: Since the subject "Here" is singular

```suggestion
  - Here is the [Video](https://youtu.be/1TsGYR1dves) and [Slide](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```
